packages/babel-parser/src/util/scope.js

```diff
@@ -124,6 +165,58 @@ export default class ScopeHandler<IScope: Scope = Scope> {
     }
   }
 
+  declarePrivateName(
+    name: string,
+    elementType: ClassElementTypes,
+    pos: number,
+  ) {
+    const classScope = this.currentClassScope();
+    let redefined = classScope.privateNames.has(name);
+
+    if (elementType & CLASS_ELEMENT_KIND_ACCESSOR) {
+      const accessor = redefined && classScope.loneAccessors.get(name);
+      if (accessor) {
+        const differences = elementType ^ accessor;
+
+        // The private name can be duplicated only if it is used by
+        // two accessors with different kind (get and set), and if
+        // they have the same placement (static or not).
+        redefined =
+          !(differences & CLASS_ELEMENT_KIND_ACCESSOR) ||
+          differences & CLASS_ELEMENT_FLAG_STATIC;
```

Comment:
This makes my head hurt. Why not split into multiple if-statements?

Reply:
Or, just compare them in two separate steps:

```js
// staticBit will be zero if both are static or both are instance.
const staticBit = (accessor & CLASS_ELEMENT_FLAG_STATIC) ^ (elementType & CLASS_ELEMENT_FLAG_STATIC);
// accessorBit will be zero if they are opposite types.
const accessorBit = (accessor & CLASS_ELEMENT_KIND_ACCESSOR) & (elementType & CLASS_ELEMENT_KIND_ACCESSOR);

redefined = !!(staticBit | accessorBit);
```

Reply:
Re-reading it made my head hurt too :stuck_out_tongue: 

I'm rewriting it to this code:
```js
        const oldStatic = accessor & CLASS_ELEMENT_FLAG_STATIC;
        const newStatic = elementType & CLASS_ELEMENT_FLAG_STATIC;

        const oldKind = accessor & CLASS_ELEMENT_FLAG_STATIC;
        const newKind = elementType & CLASS_ELEMENT_FLAG_STATIC;

        // The private name can be duplicated only if it is used by
        // two accessors with different kind (get and set), and if
        // they have the same placement (static or not).
        redefined = oldKind === newKind || oldStatic !== newStatic;
```